sdk/test-utils/recorder/src/utils.ts

```diff
@@ -113,6 +115,15 @@ export function applyReplacementMap(
     if (env[key]) {
       updated = replaceAll(updated, encodeRFC3986(env[key]!), encodeRFC3986(replacement));
       updated = replaceAll(updated, env[key]!, replacement);
+      if (
+        env[key]!.startsWith("http") &&
+        replacement.startsWith("http") &&
+        parse(env[key]!).hostname
+      ) {
+        // If an ENV variable and its replacement start with `http` with a valid hostname, replace the hostname
+        // with the one provided in the replacement. This has no effect incase the URI is already replaced in the previous step.
+        updated = replaceAll(updated, parse(env[key]!).hostname, parse(replacement).hostname);
```

Comment:
So another case to consider is if an env var contains a URL that contains a secret or just something we wouldn't want to commit to git in the path component i.e. `https://api.contoso.com/users/<my user guid>/profile`. I don't think we have anything like this currently, but I could see how something like that could eventually come up.

In that case Nock will transfer the path component to the `.get` or `.post` calls and they won't be replaced.

Maybe that's not worth doing now (we'd have to take the path component and add another replacement for it that _only_ applies to requests to that hostname), but it's something to keep in mind. This is a patch that makes it easier to avoid leaking the secret hostname, but doesn't get us full URL sanitization in the env vars.

Reply:
Thanks for the comment. Sounds good.
Though I did think about this, did not want to aggressively match each of them to avoid adverse effects of over matching which might definitely break playback.

I'm hoping these cases will be covered in the tests once we have the complete fake service which would act as the examples for all the existing/upcoming features to mock in services.

Reply:
Also, the recorder is extensible, so users can implement their own filters, if that makes sense. Use cases of the custom replacements methods will highlight the necessity of inner improvements, while also showing how necessary the changes are. This is to avoid making the core too complex, too early on.